outputs.tf

```
@@ -1,14 +1,4 @@
-output "ami_id" {
-  description = "The AMI id result of the search"
-  value = "${data.aws_ami.search.id}"
-}
-
-output "root_device_name" {
```

I think you could keep root_device_name, 
```suggestion
output "root_device_name_map" {
	  description = "The device name of the root dev"
	  value =  "${zipmap(var.os, data.aws_ami.search.*.root_device_name}"
	}
```